src/script/interpreter.cpp

```diff
@@ -1179,15 +1219,25 @@ uint256 SignatureHash(const CScript& scriptCode, const CTransaction& txTo, unsig
         // Input prevouts/nSequence (none/all, depending on flags)
         ss << hashPrevouts;
         ss << hashSequence;
+        // Outputs (none/one/all, depending on flags)
+        ss << hashOutputs;
+        if (sigversion == SIGVERSION_SAPLING) {
```

Comment:
this is always true,  we are already inside `if (sigversion == SIGVERSION_SAPLING) {}`
Maybe we should do
```cpp
if (txTo.hasSaplingData()) {}
```
?

Reply:
good catch, updated.